modules/gentemplates/openapi3-doc/src/main/java/com/reprezen/genflow/openapi3/doc/ArrayHelper.xtend

```diff
@@ -54,11 +54,11 @@ class ArrayHelper implements Helper {
 		val max = obj.maxItems
 		val min = obj.minItems
 		switch null {
-			case min === null && max === null:
+			case (min === null || min === 0) && (max === null || max === 0):
 				"[*]"
-			case min == 0 && max == 1:
+			case (min === null || min === 0) && max === 1:
 				"[?]"
-			case min == 1 && max === null:
+			case (min !== null && min === 1) && (max === null || max === 0):
 				"[+]"
 			case min !== null && max === null: '''[«min»+]'''
 			case min === null && max !== null: '''[..«max»]'''
```

Comment:
@ghillairet , @andylowry , I think I understand what was wrong here. The problem would have been on lines 59 and 61, which start with the comparisons `min == 0` and `min == 1`, respectively. Xtend binds `==` to `Object.equals()`, so that operator cannot be called safely with a null left operand.

If that's correct, your null checks seem to be the solution to this problem.

Reply:
I see. Thanks for explaining.